temporian/implementation/numpy/operators/set_index.py

```diff
@@ -5,11 +5,10 @@
 
 from temporian.core.operators.set_index import SetIndexOperator
 from temporian.implementation.numpy import implementation_lib
-from temporian.implementation.numpy.data.feature import DTYPE_REVERSE_MAPPING
 from temporian.implementation.numpy.data.event import IndexData
 from temporian.implementation.numpy.data.event import NumpyEvent
+from temporian.implementation.numpy.data.feature import DTYPE_REVERSE_MAPPING
 from temporian.implementation.numpy.operators.base import OperatorImplementation
-from temporian.implementation.numpy import implementation_lib
 
 
 class SetIndexNumpyImplementation(OperatorImplementation):
```

Comment:
Looking more at it, we have three complex and independent codes: add_index (i.e. append=True), set_index and drop_index. This is several ways to do the same thing. Given the complexity of the code, and if there is no way to factor the code, I think dropping one of them (e.g. set_index would make sense).